kong/plugins/rate-limiting/schema.lua

```diff
@@ -10,9 +10,11 @@ return {
     day = { type = "number" },
     month = { type = "number" },
     year = { type = "number" },
-    limit_by = { type = "string", enum = {"consumer", "credential", "ip"}, default = "consumer" },
+    limit_by = { type = "string", enum = {"consumer", "credential", "key", "ip"}, default = "consumer" },
     policy = { type = "string", enum = {"local", "cluster", REDIS}, default = "cluster" },
     fault_tolerant = { type = "boolean", default = true },
+    key_name = { type = "string" },
+    key_in_body = { type = "boolean", default = false },
```

Comment:
This schema should also ensure that `key_in_body` cannot be set if `key` is not the current policy?

Reply:
The `key_in_body` is invalid if `key` is not the current policy.  It will be used when policy is `key` only. 
Do you have any advice?

Reply:
Yes, I am advising that if `key_in_body` is configured while `limit_by` is not `key`, then `self_check` returns an error; because as you said, `key_in_body` is invalid if `key` is not the current limiter.

In the new schema library, this would translate with what we call "entity checkers", that can ensure that a field if set if and only if _another_ field is set a particular way. See an example [here](https://github.com/Kong/kong/blob/next/kong/plugins/rate-limiting/schema.lua#L26-L27). Of course, this PR still has to remain in a pending state until the new schemas are merged into master, at which point, the above behavior can be implemented in this PR.

Reply:
Thanks, I'll update it after the new schema is merged.